lib/presentation/mixins/handle_video_download_mixin.dart

```diff
@@ -0,0 +1,60 @@
+import 'dart:io';
+
+import 'package:fluffychat/utils/platform_infos.dart';
+import 'package:fluffychat/widgets/video_viewer_dialog.dart';
+import 'package:flutter/material.dart';
+import 'package:matrix/matrix.dart';
+import 'package:path_provider/path_provider.dart';
+import 'package:universal_html/html.dart' as html;
+
+mixin HandleVideoDownloadMixin {
+  Future<String> handleDownloadVideoEvent(
+    BuildContext context,
+    Event event,
+  ) async {
+    lastSelectedVideoEventId = event.eventId;
+    final videoFile = await event.downloadAndDecryptAttachment();
+    if (PlatformInfos.isWeb) {
+      final blob = html.Blob([videoFile.bytes]);
+      if (lastSelectedVideoEventId == event.eventId) {
+        _playVideoActionWeb(context, html.Url.createObjectUrlFromBlob(blob));
+      }
+      return html.Url.createObjectUrlFromBlob(blob);
```

Comment:
too many place did it, please create util so can reuse

Reply:
done

Reply:
where is it ?

Reply:
```
final blob = html.Blob([videoFile.bytes]);
return html.Url.createObjectUrlFromBlob(blob)
```

Reply:
good idea